app/src/main/java/de/tum/in/tumcampusapp/component/ui/search/adapter/RecentSearchesAdapter.kt

```diff
@@ -60,25 +60,36 @@ class RecentSearchesAdapter(
                     icon.setImageResource(R.drawable.ic_person)
                     icon.setBackgroundResource(R.drawable.circle_background)
                 }
-                RecentsDao.ROOMS -> {
+                RecentsDao.LECTURES -> {
                     try {
-                        val room = RoomFinderRoom.fromRecent(recentSearch)
-                        titleText.text = room.formattedAddress
+                        val lecture = Lecture.fromRecent(recentSearch)
+                        titleText.text = lecture.title
                     } catch (exception: Exception) {
                         titleText.setText(R.string.not_available_search)
                     }
-                    icon.setImageResource(R.drawable.ic_room)
+                    icon.setImageResource(R.drawable.ic_lecture)
                     icon.setBackgroundResource(R.drawable.search_result_icon_background)
                     profilePicture.setImageDrawable(null)
                 }
-                RecentsDao.LECTURES -> {
+                RecentsDao.NAVIGA_TUM_BUILDINGS -> {
                     try {
-                        val lecture = Lecture.fromRecent(recentSearch)
-                        titleText.text = lecture.title
+                        val navigationEntity = NavigationEntity.fromRecent(recentSearch)
+                        titleText.text = navigationEntity.getFormattedName()
                     } catch (exception: Exception) {
                         titleText.setText(R.string.not_available_search)
                     }
-                    icon.setImageResource(R.drawable.ic_lecture)
+                    icon.setImageResource(R.drawable.ic_building)
+                    icon.setBackgroundResource(R.drawable.search_result_icon_background)
+                    profilePicture.setImageDrawable(null)
+                }
+                RecentsDao.NAVIGA_TUM_ROOMS -> {
+                    try {
+                        val navigationEntity = NavigationEntity.fromRecent(recentSearch)
+                        titleText.text = navigationEntity.getFormattedName()
+                    } catch (exception: Exception) {
+                        titleText.setText(R.string.not_available_search)
+                    }
+                    icon.setImageResource(R.drawable.ic_room)
                     icon.setBackgroundResource(R.drawable.search_result_icon_background)
                     profilePicture.setImageDrawable(null)
```

Comment:
Since this is 2x basically the same code except for the setImageResource calls, we should simplify this and have an interiour when clause for this